modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/components/page-accessibility-tool/components/Violations.js

```diff
@@ -0,0 +1,234 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayButtonWithIcon} from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import {ClayCheckbox} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayList from '@clayui/list';
+import React, {useMemo, useState} from 'react';
+
+import Rule from './Rule';
+import {
+	TYPES,
+	useFilteredViolationsDispatch,
+	useFilteredViolationsValues,
+} from './useFilteredViolations';
+
+const IMPACT_FILTER_OPTIONS = [
+	{label: Liferay.Language.get('critical'), value: 'critical'},
+	{label: Liferay.Language.get('serious'), value: 'serious'},
+	{label: Liferay.Language.get('moderate'), value: 'moderate'},
+	{label: Liferay.Language.get('minor'), value: 'minor'},
+];
+
+const CATEGORY_FILTER_OPTIONS = [
+	{label: 'WCAG 2.0A', value: 'wcag2a'},
+	{label: 'WCAG 2.0 Level AA', value: 'wcag2aa'},
+	{label: 'WCAG 2.1 Level A', value: 'wcag21a'},
+	{label: 'WCAG 2.1 Level AA', value: 'wcag21aa'},
+	{label: 'Best Practices', value: 'best-practice'},
+];
+
+function ViolationsFilter() {
+	const [dropdownExpanded, setDropdownExpanded] = useState(false);
+	const dispatch = useFilteredViolationsDispatch();
+	const {selectedCategories, selectedImpact} = useFilteredViolationsValues();
+
+	return (
+		<>
+			<ClayDropDown
+				active={dropdownExpanded}
+				closeOnClickOutside
+				onActiveChange={setDropdownExpanded}
+				trigger={
+					<ClayButtonWithIcon
+						displayType="unstyled"
+						small
+						symbol="filter"
+					/>
+				}
+			>
+				<ClayDropDown.ItemList>
+					<ClayDropDown.Group
+						header={Liferay.Language.get('filter-by-impact')}
+					>
+						{IMPACT_FILTER_OPTIONS.map((item) => {
+							const existsImpact =
+								selectedImpact.findIndex(
+									(element) => element === item.value
+								) !== -1;
+
+							return (
+								<ClayDropDown.Section
+									active={existsImpact}
+									key={item.value}
+								>
+									<ClayCheckbox
+										active={existsImpact}
+										label={item.label}
+										onChange={() => {
+											if (existsImpact) {
+												dispatch({
+													payload: item.value,
+													type: TYPES.IMPACT_REMOVE,
+												});
+
+												return;
+											}
+
+											dispatch({
+												payload: item.value,
+												type: TYPES.IMPACT_ADD,
+											});
+										}}
+									></ClayCheckbox>
```

Comment:
In that use case, we can preferably use the self-closing tag, instead of `<ClayCheckbox></ClayCheckbox>`, especially when it does not have `children`.